service-front/module/Application/test/Controller/CPFlowControllerTest.php

```diff
@@ -269,4 +274,101 @@ public function testPostOfficeCountriesPage(): void
+            ->with($this->uuid)
+            ->willReturn($mockResponseDataIdDetails);
+
+        $this->dispatch("/$this->uuid/cp/choose-country-id", 'GET');
+        $this->assertResponseStatusCode(200);
+        $this->assertModuleName('application');
+        $this->assertControllerName(CpFlowController::class); // as specified in router's controller name alias
+        $this->assertControllerClass('CpFlowController');
+        $this->assertMatchedRouteName('root/cp_choose_country_id');
+
+        $response = $this->getResponse()->getContent();
+
+        $this->assertStringContainsString('Choose document', $response);
+    }
+
+    public function testPostOfficeCountriesIdEmptyPostErrorPage(): void
+    {
+        $mockResponseDataIdDetails = $this->returnOpgResponseData();
+
+        $this
+            ->opgApiServiceMock
+            ->expects(self::once())
+            ->method('getDetailsData')
+            ->with($this->uuid)
+            ->willReturn($mockResponseDataIdDetails);
+
+        $this->dispatch("/$this->uuid/cp/choose-country-id", 'POST', []);
+        $this->assertResponseStatusCode(200);
+        $this->assertModuleName('application');
+        $this->assertControllerName(CpFlowController::class); // as specified in router's controller name alias
+        $this->assertControllerClass('CpFlowController');
+        $this->assertMatchedRouteName('root/cp_choose_country_id');
+
+        $response = $this->getResponse()->getContent();
+
+        $this->assertStringContainsString('Please choose a type of document', $response);
+    }
+
+    public function testPostOfficeCountriesIdPostFailedValidationErrorPage(): void
+    {
+        $mockResponseDataIdDetails = $this->returnOpgResponseData();
+
+        $this
+            ->opgApiServiceMock
+            ->expects(self::once())
+            ->method('getDetailsData')
+            ->with($this->uuid)
+            ->willReturn($mockResponseDataIdDetails);
+
+        $this->dispatch(
+            "/$this->uuid/cp/choose-country-id",
+            'POST',
+            ['id_method' => 'PASSPOT']
+        );
+        $this->assertResponseStatusCode(200);
+        $this->assertModuleName('application');
+        $this->assertControllerName(CpFlowController::class); // as specified in router's controller name alias
+        $this->assertControllerClass('CpFlowController');
+        $this->assertMatchedRouteName('root/cp_choose_country_id');
+
+        $response = $this->getResponse()->getContent();
+
+        $this->assertStringContainsString('This document code is not recognised', $response);
+    }
+
+    public function testPostOfficeCountriesIdPostPage(): void
+    {
+        $mockResponseDataIdDetails = $this->returnOpgResponseData();
+
+        $this
+            ->opgApiServiceMock
+            ->expects(self::once())
+            ->method('getDetailsData')
+            ->with($this->uuid)
+            ->willReturn($mockResponseDataIdDetails);
```

Comment:
Should also test the new data is saved:
```php
$this
    ->opgApiServiceMock
    ->expects(self::once())
    ->method('updateIdMethodWithCountry')
    ->with($this->uuid, ['id_method' => 'PASSPORT']);
```
Not least because the shape of `idMethodIncludingNation` doesn't seem to be defined anywhere